futures-util/src/async_await/spawn.rs

```diff
@@ -0,0 +1,61 @@
+/// Spawns a task onto the context's executor that polls the given future with
+/// output `()` to completion.
+///
+/// This macro returns a [`Result`] that contains a
+/// [`SpawnError`](crate::task::SpawnError) if spawning fails.
+///
+/// You can use [`spawn_with_handle!`] if you want to spawn a future
+/// with output other than `()` or if you want to be able to await its
+/// completion.
+///
+/// ```
+/// #![feature(async_await, await_macro, futures_api)]
+/// #[macro_use] extern crate futures;
+/// # futures::executor::block_on(async {
+///
+/// let future = async { /* ... */ };
+/// spawn!(future).unwrap();
+/// # });
+/// ```
+#[macro_export]
+macro_rules! spawn {
+    ($future:expr) => {
+        await!($crate::future::lazy(|cx| {
+            use $crate::task::ExecutorExt;
+            cx.executor().spawn($future)
+        }))
+    }
+}
+
+/// Spawns a task onto the context's executor that polls the given future to
+/// completion and returns a future that resolves to the spawned future's
+/// output.
+///
+/// This macro returns a [`Result`] that contains a
+/// [`JoinHandle`](crate::task::JoinHandle), or, if spawning fails, a
+/// [`SpawnError`](crate::task::SpawnError).
+/// [`JoinHandle`](crate::task::JoinHandle) is a future that resolves
+/// to the output of the spawned future
+///
+/// # Examples
+///
+/// ```
+/// #![feature(async_await, await_macro, futures_api)]
+/// #[macro_use] extern crate futures;
+/// # futures::executor::block_on(async {
+/// use futures::future;
+///
+/// let future = future::ready(1);
+/// let join_handle = spawn_with_handle!(future).unwrap();
+/// assert_eq!(await!(join_handle), 1);
+/// # });
+/// ```
+#[macro_export]
+macro_rules! spawn_with_handle {
+    ($future:expr) => {
+        await!($crate::future::lazy(|cx| {
```

Comment:
`$crate::core_reexport::await!` maybe? (probably not, can just rely on await being in scope).

Reply:
In fact, it's probably a builtin macro rather than a core defined macro anyway

Reply:
The other macros (`poll!`, `pending!`) do it this way as well. 

I don't know where `await!` comes from, but I don't think it's from libcore.